### ISSUE TITLE
Add "system" option for "auth_server_scope"

### ISSUE THREAD

**Issue by tkjwa** [okta/terraform-provider-okta#887]
### Community Note

- Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
- Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
- If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

In Okta the scope object has a `system` property which is not available in the `auth_server_scope` resource: https://developer.okta.com/docs/reference/api/authorization-servers/#scope-object 
As such it is not possible to manage scopes created by Okta with terraform

### New or Affected Resource(s)

- okta_auth_server_scope

### Potential Terraform Configuration

```hcl
resource "okta_auth_server_scope" "example" {
  auth_server_id   = "<auth server id>"
  metadata_publish = "NO_CLIENTS"
  name             = "example"
  consent          = "IMPLICIT"
  system           = true # <== ALLOW THIS
}
```

Thank you

**Comment by bogdanprodan-okta** [comment]
Hi, @tkjwa! Thanks for submitting this issue! As far as I know, the `system` property is read-only and is just indicates whether Okta created this scope.

**Comment by tkjwa** [comment]
Hello @bogdanprodan-okta ! Indeed but it needs to be sent to the api when you want to update a scope created by okta.

I've made tests with direct api calls: `PUT /api/v1/authorizationServers/ausabc123/scopes/scpabc123`
with the following body: `{"consent": "REQUIRED", "name": "profile", "metadataPublish":"ALL_CLIENTS"}` I have an error, but it works with `{"consent": "REQUIRED", "name": "profile", "metadataPublish":"ALL_CLIENTS", "system": true}`

So it looks like even if it can't be updated it needs to be in the request for "system" scopes

**Comment by tkjwa** [comment]
Please don't close this issue, in the current state it is not possible to manage scopes created by okta with terraform

**Comment by monde** [comment]
Thanks @tkjwa I'll be picking this up and will look into it.

**Comment by tkjwa** [comment]
:+1:

**Comment by monde** [comment]
@tkjwa just double checking, you want to have `system` exposed as an attribute, correct? In the Okta API that property is read-only. It would be listed as an attribute here https://registry.terraform.io/providers/okta/okta/latest/docs/resources/auth_server_scope#attributes-reference

Investigating with postman making API calls against the Okta API against the org we use for the provider's testing:

`GET /api/v1/authorizationServers/{authorizationServerId}/scopes/{scopeId}`
```
{
    "id": "{scopeId}",
    "name": "Test",
    "displayName": "Test",
    "description": "Test",
    "system": false,
    "metadataPublish": "ALL_CLIENTS",
    "consent": "REQUIRED",
    "apiResourceId": null,
    "default": false,
    "_links": {
        "self": {
            "href": "https://{org}/api/v1/authorizationServers/{authorizationServerId}/scopes/{scopeId}",
            "hints": {
                "allow": [
                    "GET",
                    "PUT",
                    "DELETE"
                ]
            }
        }
    }
}
```


`PUT /api/v1/authorizationServers/{authorizationServerId}/scopes/{scopeId}`

body
```
{
  "description": "Test, Test",
  "name": "Test",
  "consent": "REQUIRED",
  "metadataPublish": "ALL_CLIENTS",
  "system": true
}
```

Response if 400
```
{
    "errorCode": "E0000001",
    "errorSummary": "Api validation failed: system",
    "errorLink": "E0000001",
    "errorId": "oae_oK04ZcdTj2tU13S1kUUTA",
    "errorCauses": [
        {
            "errorSummary": "system: 'system' cannot be modified."
        }
    ]
}
```

So the resource would be this, `system` is not listed in the body:

```terraform
resource "okta_auth_server_scope" "example" {
  auth_server_id   = "<auth server id>"
  metadata_publish = "NO_CLIENTS"
  name             = "example"
  consent          = "IMPLICIT"
}
```
and you can access elsewhere in your config as `okta_auth_server_scope.example.system`

**Comment by tkjwa** [comment]
Hi @monde 

> you want to have `system` exposed as an attribute, correct? In the Okta API that property is read-only. It would be listed as an attribute here [https://registry.terraform.io/providers/okta/okta/latest/docs/resources/auth_server_scope#attributes-reference](https://registry.terraform.io/providers/okta/okta/latest/docs/resources/auth_server_scope#attributes-reference)

That's correct !

Regarding the 400 error, it is impossible to change the "system" attribute, so there could be some kind of validation after the scope is in the state

What I want to achieve is:

*   import a sytem scope (created by okta with the auth server) which has a system=true attribute
*   then i want to update this scope using TF (not update the system attribute itself which is forbidden)

> and you can access elsewhere in your config as `okta_auth_server_scope.example.system`

That's not required for me but it should be available indeed if it gets implemented

Thank you

**Comment by tkjwa** [comment]
Thank you @monde !

**Comment by monde** [comment]
> Thank you @monde !

This will be in Friday's release v3.28.0, thanks

**Comment by monde** [comment]
Released as [v3.28.0](https://github.com/okta/terraform-provider-okta/releases/tag/v3.28.0).
